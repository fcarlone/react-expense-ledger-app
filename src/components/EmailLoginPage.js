import React from 'react';
import { connect } from 'react-redux';
import Header from '../components/Header'
import { createUser, loginUser } from '../actions/auth';

export class EmailLoginPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      error: ''
    };
  };

  onEmailChange = (e) => {
    e.persist();
    this.setState(() => ({ email: e.target.value }))
  };

  onChangePassword = (e) => {
    e.persist();
    this.setState(() => ({ password: e.target.value }))

  };

  onSubmit = (e) => {
    e.preventDefault()

    if (!this.state.email || !this.state.password) {
      this.setState(() => ({
        error: 'Try logging in again.'
      }))
    };
    console.log(`e:{}`)
  };

  onLogin = () => {
    console.log(`onLogin ${this.state.email} ${this.state.password}`);
    this.props.loginUser(this.state.email, this.state.password);
  }

  onSignUp = () => {
    console.log(`onSignUp ${this.state.email}, ${this.state.password}`)
    this.props.createUser(this.state.email, this.state.password);
  };

  render() {
    return (
      <header className="header">
        <Header />
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Log in with your email account</h1>
            <div className="page-header__actions">
              <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">Error: {this.state.error}</p>}
                <input
                  type="text"
                  placeholder="email address"
                  className="text-input"
                  autoFocus
                  value={this.state.email}
                  onChange={this.onEmailChange}
                />
                <input
                  type="text"
                  placeholder="password"
                  className="text-input"
                  value={this.state.password}
                  onChange={this.onChangePassword}
                />
                <div>
                  <button className="button-layout-login" onClick={this.onLogin}>Log In</button>
                  <button className="button-layout-signup" onClick={this.onSignUp}>Sign Up</button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </header>
    )
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    createUser: (email, password) => dispatch(createUser(email, password)),
    loginUser: (email, password) => dispatch(loginUser(email, password))
  };
};

export default connect(undefined, mapDispatchToProps)(EmailLoginPage);
