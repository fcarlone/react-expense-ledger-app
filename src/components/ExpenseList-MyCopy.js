import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';

// stateless functional component
const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    <p>{props.filters.text}</p>
    <p>{props.expenses.length}</p>
    <ExpenseListItem />
  </div>
);

// Higher Order Function
const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filters: state.filters
  };
};

// const ConnectedExpenseList = connect((state) => {
//   return {
//     // name: 'Frank'
//     expenses: state.expenses
//   };
// })(ExpenseList);

export default connect(mapStateToProps)(ExpenseList);

// export default ExpenseList;
// export default ConnectedExpenseList;
