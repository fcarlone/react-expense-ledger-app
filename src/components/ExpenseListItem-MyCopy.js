// Export a stateless functional component
// description, amount, createdAt

// import React and Connect
import React from 'react';
import { connect } from 'react-redux';

// Refactor Code 

// Stateless Functional Component
const ExpenseListItem = (props) => (
  <div>
    <h1>ExpenseListItem</h1>
    <p>{props.expenses}</p>
  </div>
)

// Higher Order Component
const MapStateToProps = (state) => {
  return {
    expenses: state.expenses.map((expense) => {
      return (
        <li key={expense.id}>Description: {expense.description} Amount: {expense.amount} Created At: {expense.createdAt}</li>
      )
    })
  }
}

export default connect(MapStateToProps)(ExpenseListItem);

// Original Code

// const ExpenseListItem = (props) => (
//   <div>
//     <h1>ExpenseListItem</h1>
//     <p>{props.expenses}</p>
//   </div>
// )

// const ConnectExpenseListItem = connect((state) => {
//   return {
//     expenses: state.expenses.map((expense) => {
//       return (
//         <li key={expense.id}>Description: {expense.description} Amount: {expense.amount} Created At: {expense.createdAt}</li>
//       )
//     })
//   }
// })(ExpenseListItem);

// export default ConnectExpenseListItem;
