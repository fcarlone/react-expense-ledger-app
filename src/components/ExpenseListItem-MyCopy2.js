// Export a stateless functional component
// description, amount, createdAt

// import React
import React from 'react';

// Stateless Functional Component
// Destructure the props object
const ExpenseListItem = ({ description, amount, createdAt }) => (
  <div>
    <h3>{description}</h3>
    <p>{amount} - {createdAt}</p>
  </div>
)

export default ExpenseListItem;

// // Higher Order Component
// const MapStateToProps = (state) => {
//   return {
//     expenses: state.expenses.map((expense) => {
//       return (
//         <li key={expense.id}>Description: {expense.description} Amount: {expense.amount} Created At: {expense.createdAt}</li>
//       )
//     })
//   }
// }

// export default connect(MapStateToProps)(ExpenseListItem);

// // Original Code

// // const ExpenseListItem = (props) => (
// //   <div>
// //     <h1>ExpenseListItem</h1>
// //     <p>{props.expenses}</p>
// //   </div>
// // )

// // const ConnectExpenseListItem = connect((state) => {
// //   return {
// //     expenses: state.expenses.map((expense) => {
// //       return (
// //         <li key={expense.id}>Description: {expense.description} Amount: {expense.amount} Created At: {expense.createdAt}</li>
// //       )
// //     })
// //   }
// // })(ExpenseListItem);

// // export default ConnectExpenseListItem;
