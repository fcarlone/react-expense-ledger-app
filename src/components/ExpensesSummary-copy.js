import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => {
  return (
    <div>
      <p>Expenses Summary</p>
      {
        props.expensesLength.length !== 0 ? (
          <p>Viewing {props.expensesLength.length} {expenseWord} totalling {numeral(props.expensesAmount / 100).format('$0,0.00')}</p>
        ) : (
            <p></p>
          )
      }
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    expensesLength: selectExpenses(state.expenses, state.filters),
    expensesAmount: selectExpensesTotal(state.expenses)
  };
}


export default connect(mapStateToProps)(ExpenseSummary);
