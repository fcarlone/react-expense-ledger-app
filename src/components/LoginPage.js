import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">React Expense Ledger</h1>
      <p>It's time to get your expenses under control</p>
      <button className="button-layout" onClick={startLogin}>Login with Google</button>
      <Link className="button-layout-2" to="/login">Login with Email Address</Link>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
