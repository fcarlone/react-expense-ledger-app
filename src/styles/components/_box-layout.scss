.box-layout {
  align-items: center;
  background: url('/images/rel3.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(#ccc3c3, .15);
  border-radius: 3px;
  padding: $l-size $m-size;
  text-align: center;
  width: 37rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}

.box-layout__subtitle {
  font-weight: 600
}
