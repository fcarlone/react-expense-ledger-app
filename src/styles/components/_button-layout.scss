.button-layout {
  background-color: $light-blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button-layout-2 {
  background-color: white;
  border: none;
  border-radius: 5px;
  color: black;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 500;
  line-height: 1;
  margin-top: $m-size;
  padding: $s-size;
  text-decoration: none;
}

.button-layout-2-icon {
  padding-right: 5px;
}

.button-layout-login {
  background-color: #1c88bf;
  border: none;
  border-radius: 10px;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 500;
  line-height: 1;
  margin-top: $m-size;
  padding: $s-size;
  text-decoration: none;
}

.button-layout-signup {
  background-color: #1c88bf;
  border: none;
  border-radius: 10px;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 500;
  line-height: 1;
  margin-left: $m-size;
  margin-top: $m-size;
  padding: $s-size;
  text-decoration: none;
}

.button-layout-login:hover,
.button-layout-signup:hover {
  background: grey;
  color: white;
}

.button-layout--link {
  background-color: $dark-blue;
  border: none;
  color: white;
}

.button-layout--secondary {
  background-color: #888888;
}

.google-button {
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 500;
  line-height: 1;
  margin-top: $m-size;
  padding: $s-size;
  text-decoration: none
}

.google-button-icon {}

.google-button {
  text-align: center;
  border-radius: 5px;
}

.google-button:hover {
  color: $light-blue;
}

.button-layout-2-text:hover {
  color: $light-blue;
}

svg {
  width: $m-size;
}
