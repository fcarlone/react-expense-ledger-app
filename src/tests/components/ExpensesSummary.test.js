import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpensesSummary';


test('should correctly render 2 expenses totalling $50.00', () => {
  const wrapper = shallow(<ExpenseSummary expensesLength={2} expensesAmount={5000} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render 1 expense totalling $20.00', () => {
  const wrapper = shallow(<ExpenseSummary expensesLength={1} expensesAmount={2000} />);
  expect(wrapper).toMatchSnapshot();
})

