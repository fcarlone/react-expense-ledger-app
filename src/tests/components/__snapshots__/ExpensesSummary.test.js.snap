// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render 1 expense totalling $20.00 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button-layout"
        replace={false}
        to="/create"
      >
        Add Expenses
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render 2 expenses totalling $50.00 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button-layout"
        replace={false}
        to="/create"
      >
        Add Expenses
      </Link>
    </div>
  </div>
</div>
`;
